addons/point_of_sale/views/account_statement_view.xml

```diff
@@ -19,4 +19,15 @@
                 </xpath>
             </field>
         </record>
+
+        <record id="view_bank_statement_pos_session" model="ir.ui.view">
+            <field name="name">view.bank.statement.pos.session</field>
+            <field name="model">account.bank.statement</field>
+            <field name="inherit_id" ref="account.view_bank_statement_form"/>
```

Comment:
As it is really specific to pos, maybe it is safer to make this view primary, with the lowest priority, remove the button, instead of making it invisible on context. And set a context on statement_ids containing key 'form_view_ref' to 'point_of_sale.view_bank_statement_pos_session'

P.S.: making it 'primary' will not override the existing one, it will just use the existing one, applying your xpath and it will be a new view.
Putting the lowest priority (I think it's 1000 or something like that) will let the one defined in the accounting as the default one, and this one will never be used except if it is reference in a form_view_ref. (or in an action)